.github/workflows/daily_fuzz.yaml

```diff
@@ -0,0 +1,72 @@
+name: Daily parser fuzz
+
+on:
+  workflow_dispatch:
+  schedule:
+    - cron: "0 0 * * *"
+  pull_request:
+    paths:
+      - ".github/workflows/daily_fuzz.yaml"
+
+permissions:
+  contents: read
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.head_ref || github.run_id }}
+  cancel-in-progress: true
+
+env:
+  CARGO_INCREMENTAL: 0
+  CARGO_NET_RETRY: 10
+  CARGO_TERM_COLOR: always
+  RUSTUP_MAX_RETRIES: 10
+  PACKAGE_NAME: ruff
+  FORCE_COLOR: 1
+
+jobs:
+  fuzz:
+    name: Fuzz
+    runs-on: ubuntu-latest
+    timeout-minutes: 20
+    # Don't run the cron job on forks:
+    if: ${{ github.repository == 'astral-sh/ruff' || github.event_name != 'schedule' }}
+    steps:
+      - uses: actions/checkout@v4
+      - uses: actions/setup-python@v5
+        with:
+          python-version: "3.12"
+      - name: Install uv
+        run: curl -LsSf https://astral.sh/uv/install.sh | sh
+      - name: Install Python requirements
+        run: uv pip install -r scripts/fuzz-parser/requirements.txt --system
+      - name: "Install Rust toolchain"
+        run: rustup show
+      - name: "Install mold"
+        uses: rui314/setup-mold@v1
+      - uses: Swatinem/rust-cache@v2
+      - name: Build ruff
+        # A debug build means the script runs slower once it gets started,
+        # but this is outweighed by the fact that a release build takes *much* longer to compile in CI
+        run: cargo build --locked
+      - name: Fuzz
+        run: python scripts/fuzz-parser/fuzz.py $(shuf -i 0-1000000 -n 1000) --test-executable target/debug/ruff
```

Comment:
Is this clamping the random numbers to a range between 1000000 is there a reason why not to go higher (or move the random number generation into the fuzzer script?

Reply:
> Is this clamping the random numbers to a range between 1000000

yes

> is there a reason why not to go higher

It looks like we can go as high as 9999999999999999999! Add any more digits to that number and the `shuf` shell utility fails (at least for me locally):

```
(ruff) % echo $(shuf -i0-99999999999999999991 -n 100)                                                                                                                     ~/dev/ruff
shuf: invalid input range: ‘0-99999999999999999991’: Value too large to be stored in data type
```

I'll bump the number to the maximum.

> move the random number generation into the fuzzer script?

The argparsing gets significantly more complex if we move the random generation into the fuzzer script, so I'd _prefer_ to simply pass a randomly selected set of seeds from the command line.